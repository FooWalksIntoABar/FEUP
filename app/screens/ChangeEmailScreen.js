import React from 'react'
import { View, Text, TextInput, TouchableOpacity } from 'react-native'

import PropTypes from 'prop-types'

import styles from '../styles/ParentStyles/ChangeEmailScreen.style'

export default class ChangeEmailScreen extends React.Component {
  static navigationOptions = {
    headerMode: 'screen',
    navigationOptions: {
      headerTitle: 'Repor E-mail'
    }
  };
  render () {
    return (
      <View style={styles.generalLayout}>
        <View style={styles.container}>
          <Text>E-mail Atual</Text>
          <Text>{this.props.navigation.state.params.email}</Text>
        </View>
        <View style={styles.container}>
          <Text>Novo E-mail</Text>
          <TextInput
            keyboardType='email-address'
            placeholder='Insira o seu novo e-mail'
            style={styles.inputContainer}
          />
        </View>
        <View style={styles.container}>
          <Text>Password</Text>
          <Text>Insira a sua password para confirmar a alteração de e-mail.</Text>
          <TextInput
            secureTextEntry
            placeholder='Insira o seu novo e-mail'
            style={styles.inputContainer}
          />
        </View>
        <View >
          <TouchableOpacity style={styles.confirmBtn} onPress={() => { this.props.navigation.state.params.setEmail('hahahha') }}>
            <Text>Confirmar</Text>
          </TouchableOpacity>
        </View>
      </View>
    )
  }
}

ChangeEmailScreen.propTypes = {
  navigation: PropTypes.object.isRequired
}
