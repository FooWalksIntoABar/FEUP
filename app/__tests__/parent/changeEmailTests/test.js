import React from 'react'
import { TouchableOpacity } from 'react-native'
import { configure, shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'

import ChangeEmailScreen from '../../../screens/parent/ChangeEmailScreen'

configure({ adapter: new Adapter() })

describe('ChangeEmailScreen snapshot', () => {
  it('renders ChangeEmailScreen correctly', () => {
    const wrapper = shallow(<ChangeEmailScreen
      navigation={{ navigate: jest.fn(), state: { params: { email: 'email@aleatorio.pt', setEmail: jest.fn() } } }} />)
    expect(toJson(wrapper)).toMatchSnapshot()
    wrapper.find(TouchableOpacity).at(0).props().onPress()
    wrapper.unmount()
  })
})
