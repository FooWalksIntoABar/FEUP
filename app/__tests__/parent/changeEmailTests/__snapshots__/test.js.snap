// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangeEmailScreen snapshot renders ChangeEmailScreen correctly 1`] = `
<Component
  style={
    Object {
      "padding": 60,
    }
  }
>
  <Component
    style={
      Object {
        "paddingBottom": 20,
        "paddingTop": 20,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      E-mail Atual
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "paddingTop": 20,
          "textAlign": "center",
        }
      }
    >
      email@aleatorio.pt
    </Text>
  </Component>
  <Component
    style={
      Object {
        "paddingBottom": 20,
        "paddingTop": 20,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Novo E-mail
    </Text>
    <TextInput
      allowFontScaling={true}
      keyboardType="email-address"
      placeholder="Insira o seu novo e-mail"
      style={
        Object {
          "borderColor": "white",
          "borderRadius": 3,
          "borderWidth": 1,
          "height": 50,
          "marginTop": 20,
          "padding": 10,
        }
      }
    />
  </Component>
  <Component
    style={
      Object {
        "paddingBottom": 20,
        "paddingTop": 20,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Password
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Insira a sua password para confirmar a alteração de e-mail.
    </Text>
    <TextInput
      allowFontScaling={true}
      placeholder="Insira o seu novo e-mail"
      secureTextEntry={true}
      style={
        Object {
          "borderColor": "white",
          "borderRadius": 3,
          "borderWidth": 1,
          "height": 50,
          "marginTop": 20,
          "padding": 10,
        }
      }
    />
  </Component>
  <Component>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "orange",
          "borderRadius": 10,
          "height": 40,
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Confirmar
      </Text>
    </TouchableOpacity>
  </Component>
</Component>
`;
